src/Cooked/MockChain/GenerateTx.hs

```diff
@@ -226,15 +217,48 @@ txOutRefsToTxInsReference =
     )
     . mapM Ledger.toCardanoTxIn
 
--- Convert a list of 'Api.TxOutRef' into a 'Cardano.TxInsCollateral'
-txOutRefsToTxSkelInsCollateral :: [Api.TxOutRef] -> TxGen (Cardano.TxInsCollateral Cardano.ConwayEra)
-txOutRefsToTxSkelInsCollateral =
-  throwOnToCardanoError "txOutRefsToTxInCollateral"
-    . fmap toTxInsCollateral
-    . mapM Ledger.toCardanoTxIn
-  where
-    toTxInsCollateral [] = Cardano.TxInsCollateralNone
-    toTxInsCollateral ins = Cardano.TxInsCollateral Cardano.AlonzoEraOnwardsConway ins
+-- Computes the collateral triplet from the fee and the collateral inputs in
+-- the context
+toCollateralTriplet ::
+  TxGen
+    ( Cardano.TxInsCollateral Cardano.ConwayEra,
+      Cardano.TxTotalCollateral Cardano.ConwayEra,
+      Cardano.TxReturnCollateral Cardano.CtxTx Cardano.ConwayEra
+    )
+toCollateralTriplet = do
+  knownTxOuts <- asks managedTxOuts
+  returnCollateralWallet <- asks returnCollateralWallet
+  networkId <- asks (Emulator.pNetworkId . params)
+  collateralInsList <- asks (Set.toList . collateralIns)
+  collateralInsValue <-
+    let collateralInsResolved = mapMaybe (`Map.lookup` knownTxOuts) collateralInsList
+     in if length collateralInsResolved /= length collateralInsList
+          then throwOnString "toCollateralTriplet: unresolved txOutRefs"
+          else return $ mconcat (Api.txOutValue <$> collateralInsResolved)
```

Comment:
I thought it would be an improvement but it is actually pretty neutral. Leaving that here just in case you prefer.
```suggestion
  collateralInsValue <- do
    let collateralInsResolved = mapMaybe (`Map.lookup` knownTxOuts) collateralInsList
    unless (length collateralInsResolved == length collateralInsList) $
      throwOnString "toCollateralTriplet: unresolved txOutRefs"
    return $ mconcat (Api.txOutValue <$> collateralInsResolved)
```

Reply:
I like it.